src/test/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotTest.java

```diff
@@ -945,6 +946,75 @@ public void verifyConnectorFailsIfMarkSnapshotDoneFails(boolean consistentSnapsh
         YugabyteDBSnapshotChangeEventSource.FAIL_WHEN_MARKING_SNAPSHOT_DONE = false;
     }
 
+    @Test
+    public void snapshotShouldBeCompletedOnParentIfSplitHappenedAfterStreamCreation() throws Exception {
+        /*
+        * The objective of the test is to verify that snapshot takes place on the parent tablet if a split
+        * happens after a consistent stream is created and before connector has been deployed. Additionally,
+        * any DMLs performed after the split are not part of snapshot records.
+         */
+        TestHelper.dropAllSchemas();
+        TestHelper.execute("CREATE TABLE t1 (id INT PRIMARY KEY, name TEXT) SPLIT INTO 1 TABLETS;");
+        TestHelper.execute("CREATE TABLE t2 (id INT PRIMARY KEY, name TEXT) SPLIT INTO 1 TABLETS;");
+        TestHelper.execute("CREATE TABLE t3 (id INT PRIMARY KEY, name TEXT) SPLIT INTO 1 TABLETS;");
+
+        int recordsCount = 10;
+        String insertFormat = "INSERT INTO t1 VALUES (%d, 'value for split table');";
+        for (int i = 0; i < recordsCount; ++i) {
+            TestHelper.execute(String.format(insertFormat, i));
+        }
+
+        String dbStreamId = TestHelper.getNewDbStreamId("yugabyte", "t1", false, true, true, true);
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("public.t1", dbStreamId);
+        configBuilder.with(YugabyteDBConnectorConfig.SNAPSHOT_MODE, "initial");
+
+        YBClient ybClient = TestHelper.getYbClient(getMasterAddress());
```

Comment:
@vaibhav-yb should we use try with resources here?

Reply:
Since this is a test, it's okay to keep it like this.